pilot/pkg/networking/core/gateway_test.go

```diff
@@ -4442,3 +4442,43 @@ func TestGatewayHCMInternalAddressConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestListenerTransportSocketConnectTimeoutForGateway(t *testing.T) {
+	wantTimeout := durationpb.New(defaultGatewayTransportSocketConnectTimeout).GetSeconds()
```

Comment:
Can you convert this to a table driven test?

Reply:
Typically yes, but putting it in table form now will make adding new test cases easier 🙂